src/Data/Vec/Base.agda

```diff
@@ -64,29 +64,27 @@ iterate : (A → A) → A → ∀ {n} → Vec A n
 iterate s z {zero}  = []
 iterate s z {suc n} = z ∷ iterate s (s z)
 
-insert : Vec A n → Fin (suc n) → A → Vec A (suc n)
-insert xs       zero     v = v ∷ xs
-insert (x ∷ xs) (suc i)  v = x ∷ insert xs i v
+insertAt : Vec A n → Fin (suc n) → A → Vec A (suc n)
+insertAt xs       zero     v = v ∷ xs
+insertAt (x ∷ xs) (suc i)  v = x ∷ insertAt xs i v
 
-remove : Vec A (suc n) → Fin (suc n) → Vec A n
-remove (_ ∷ xs)     zero     = xs
-remove (x ∷ y ∷ xs) (suc i)  = x ∷ remove (y ∷ xs) i
+removeAt : Vec A (suc n) → Fin (suc n) → Vec A n
+removeAt (_ ∷ xs)     zero     = xs
+removeAt (x ∷ y ∷ xs) (suc i)  = x ∷ removeAt (y ∷ xs) i
```

Comment:
```agda
removeAt (x ∷ xs@(_ ∷ _)) (suc i) = x ∷ removeAt xs i
```

Reply:
Do we even need to match against `(_ ∷ _)` here? Would `removeAt (x ∷ xs) (suc i) = x ∷ removeAt xs i` work?

Reply:
All the better if so!

Reply:
You probably need to match for Agda to see that the recursive call is legal (it needs
to be performed on a `Fin (suc _)`).

Reply:
Thanks!